src/python/WMCore/Database/ExecuteDAO.py

```diff
@@ -160,16 +160,22 @@ def __init__(self, logger=None, configFile=None,
             config.CoreDatabase.dialect = getBackendFromDbURL(os.getenv("DATABASE"))
             config.CoreDatabase.socket = os.getenv("DBSOCK", None)
 
+        # always try to determine the dilect from the URL
+        if getattr(config.CoreDatabase, "connectUrl", None):
+            config.CoreDatabase.dialect = getBackendFromDbURL(config.CoreDatabase.connectUrl)
+
+        # finally if no socket is provided, set it to None and let WMInit to create it.
+        config.CoreDatabase.socket = getattr(config.CoreDatabase, "socket", None)
+
         # check if all database connection parameters are provided:
         if not all([getattr(config.CoreDatabase, "connectUrl", None),
```

Comment:
This is Okay, but again fragile in case you have a mysql connection url without the socket defined.

Reply:
Additional check for this could be put here, but since the possibly dialects identifiers are more than a single string, I'd say we leave it blank and let the next method `setDatabaseConnection` deal with it as it already has the option to accept the socket as `None` by default